glass/observations.py

```diff
@@ -328,3 +330,192 @@ def tomo_nz_gausserr(
     binned_nz *= nz
 
     return binned_nz  # type: ignore[no-any-return]
+
+
+class angular_variable_depth_mask:
+    r"""Variable depth mask for tomographic bins.
+
+    This class allows to create a mask with a different variable depth mask in the angular direction for each tomographic bin.
+
+    Parameters
+    ----------
+    vardepth_map : array_like
+        Map of variable which traces the depth per tomographic bin.
+    n_bins : int
+        Number of tomographic bins.
+    zbins : array_like
+        Shell redshift limits.
+    """
+
+    def __init__(self, vardepth_map, n_bins, zbins):
+        self.vardepth_map = vardepth_map
+        self.n_bins = n_bins
+        self.zbins = zbins
```

Comment:
The actual types I'll leave to you, but can you change this to be properly typed rather than specifying in the docstring? Have a look around the code base and you'll see this is used consistently throughout.

```suggestion
    vardepth_map
        Map of variable which traces the depth per tomographic bin.
    n_bins
        Number of tomographic bins.
    zbins
        Shell redshift limits.
    """

    def __init__(
    	self,
    	vardepth_map: NDArray[np.float_],
    	n_bins: int,
    	zbins: NDArray[np.float_]
  	):
        self.vardepth_map = vardepth_map
        self.n_bins = n_bins
        self.zbins = zbins
```